### PR TITLE
Euclidean algorithm in coconut

### PR DESCRIPTION
**Pull request by Amaras**
Here is the euclidean algorithm as one could implement it in the Coconunt programming language.

It uses pattern matching, and Coconut optimizes tail call recursion, so the algorithm will not exceed max recursion depth.

As it's the first time someone seems to have implemented an AAA algorithm in this language, there are a few things you need to know about it.

First (some advertising for Coconut): Coconut is compiled language which is a strict superset of Python 3: all valid Python code is valid Coconut code. The Coconut compiler compiles to Python. Coconut also adds more functional programming features built-in than Python does. You can find their website at: http://coconut-lang.org, and docs at https://coconut.readthedocs.io/en/master/DOCS.html

Second: Installation instructions: `pip install coconut` that's all.

Third (and finally): Compilation instructions: `coconut yourfile.coco` compiles to the `yourfile.py` file, which you can run with your Python 3 installation. You can also do `coconut -r yourfile`, which runs the Python file after compilation (it also keeps the file on your system). I can add the file if needed, but it is 

If you have any question, feel free to ask.

### CODE REVIEW

File: contents/euclidean_algorithm/code/cocount/euclidean.coco [hunk 1]
```diff
@@ -0,0 +1,20 @@
+def euclid_sub(a is int, 0) = a
+addpattern def euclid_sub(0, b is int) = b
+
+addpattern def euclid_sub(a is int, b is int):
+	if a == b:
+		return a
+	elif a < b:
+		return euclid_sub(a, b - a)
+	elif b < a:
+		return euclid_sub(a - b, b)
```
Comment (berquist) [inline contents/euclidean_algorithm/code/cocount/euclidean.coco:10]: Can you order this like the Python implementation, where there's implicit fallthrough for the `a == b` case?

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 2]
```diff
@@ -81,6 +81,8 @@ The algorithm is a simple way to find the *greatest common divisor* (GCD) of two
 
 {% sample lang="ps1" %}
 [import:1-14, lang="powershell"](code/powershell/euclidean_algorithm.ps1)
+{% sample lang="coco" %}
+[import:1-11, lang="coconut"](code/coconut/euclidean.coco)
```
Comment (berquist) [inline contents/euclidean_algorithm/euclidean_algorithm.md:85]:
```suggestion
[import:1-10, lang="coconut"](code/coconut/euclidean.coco)
```
(assuming you don't do the case reordering)

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 3]
```diff
@@ -169,6 +171,8 @@ Modern implementations, though, often use the modulus operator (%) like so
 
 {% sample lang="ps1" %}
 [import:16-27, lang="powershell"](code/powershell/euclidean_algorithm.ps1)
+{% sample lang="coco" %}
+[import:13-21, lang="coconut"](code/coconut/euclidean.coco)
```
Comment (berquist) [inline contents/euclidean_algorithm/euclidean_algorithm.md:175]:
```suggestion
[import:13-16, lang="coconut"](code/coconut/euclidean.coco)
```

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 4]
```diff
@@ -169,6 +171,8 @@ Modern implementations, though, often use the modulus operator (%) like so
 
 {% sample lang="ps1" %}
 [import:16-27, lang="powershell"](code/powershell/euclidean_algorithm.ps1)
+{% sample lang="coco" %}
+[import:13-16, lang="coconut"](code/coconut/euclidean.coco)
```
Comment (berquist) [inline contents/euclidean_algorithm/euclidean_algorithm.md:175]:
```suggestion
[import:12-15, lang="coconut"](code/coconut/euclidean.coco)
```

**Comment by Amaras** [comment]
Thanks for the review.
It's mainly Python, so 4 spaces, but I forgot to check that it was indenting using spaces.
I'll check with my current and future PRs for Coconut code.

**Comment by berquist** [comment]
Can you make the last Markdown change?

**Comment by Amaras** [comment]
Yep, I forgot to take that into account. Probably done by now.

**Comment by berquist** [comment]
There is something else I missed. You should modify https://github.com/algorithm-archivists/algorithm-archive/blob/master/book.json to have an entry for Coconut in the language picker. Right now, it shows as `COCO`, which you can override with the `Name` field.

**Comment by Amaras** [comment]
Yeah, I didn't know how to do that. Should be good now, if I did it right.

**Pull request merged**